src/query/functions/linear/histogram_quantile.go

```diff
@@ -339,14 +339,20 @@ func processValidQuantile(
 			buckets := b.buckets
 			// clear previous bucket values.
 			bucketValues = bucketValues[:0]
+			max := math.Inf(-1)
 			for _, bucket := range buckets {
 				// Only add non-NaN values to contention for the calculation.
 				val := values[bucket.idx]
 				if !math.IsNaN(val) {
+					if val >= max {
```

Comment:
This logic might make a little more sense inside the [sanitizeBuckers](https://github.com/m3db/m3/blob/master/src/query/functions/linear/histogram_quantile.go#L196) method, so that it doesn't have to run per query

Reply:
Can we also add a unit test as well as the integration test?

Reply:
`sanitizeBuckets` cannot be used here because it operates on series metadata level, and this transformation needs to be done on the actual datapoints. So I have extracted the logics into a new `func ensureMonotonic`, which makes it more unit-testable.
We will get ITs for this (and many other cases) once we merge https://github.com/m3db/m3/pull/2361 (and uncomment the fixed cases).